qa/tasks/vstart_runner.py

```diff
@@ -418,13 +427,92 @@ def get_keyring_path(self):
         # FIXME maybe should add a pwd arg to teuthology.orchestra so that
         # the "cd foo && bar" shenanigans isn't needed to begin with and
         # then we wouldn't have to special case this
-        return self.client_remote.run(
-            args, wait=wait, cwd=self.mountpoint
-        )
+        return self.client_remote.run(args, wait=wait, cwd=self.mountpoint,
+                                      stdin=stdin, check_status=check_status,
+                                      omit_sudo=omit_sudo)
+
+    def run_as_user(self, args, user, pwd=None, wait=True, check_status=True):
+        # FIXME maybe should add a pwd arg to teuthology.orchestra so that
+        # the "cd foo && bar" shenanigans isn't needed to begin with and
+        # then we wouldn't have to special case this
+        if str(type(args)).find('str') != -1:
+            if (args.find("'") != -1 or args.find('"') != -1):
+                raise RuntimeError("Don't surround command by single/double "
+                                   "quotes")
+            args = ['sudo', '-u', user, '-s', '/bin/bash', '-c', args]
+        elif str(type(list)).find('list') != -1:
+            for arg in args:
+                if arg.find("'") != -1 or arg.find('"') != -1:
+                    raise RuntimeError("Don't surround command by single/double "
+                                       "quotes")
+            args = ['sudo', '-u', user, '-s', '/bin/bash', '-c']
+            args.extend(args)
+
+        return self.client_remote.run(args, wait=wait, cwd=self.mountpoint,
+                                      check_status=check_status, stdin=pwd,
+                                      omit_sudo=False)
+
+    def run_as_root(self, args, wait=True, check_status=True, stdin=None):
+        # FIXME maybe should add a pwd arg to teuthology.orchestra so that
+        # the "cd foo && bar" shenanigans isn't needed to begin with and
+        # then we wouldn't have to special case this
+        if str(type(args)).find('str') != -1:
+            args = 'sudo ' + args
+        elif str(type(args)).find('list') != -1:
+            args.insert(0, 'sudo')
+
+        return self.client_remote.run(args, wait=wait, cwd=self.mountpoint,
+                                      check_status=check_status,
+                                      omit_sudo=False)
+
+    def testcmd(self, args, wait=True, stdin=None, check_status=False, omit_sudo=True):
+        # FIXME maybe should add a pwd arg to teuthology.orchestra so that
+        # the "cd foo && bar" shenanigans isn't needed to begin with and
+        # then we wouldn't have to special case this
+        return self.client_remote.run(args, wait=wait, cwd=self.mountpoint,
+                                      stdin=stdin, check_status=check_status,
+                                      omit_sudo=omit_sudo)
+
+    def testcmd_as_user(self, args, user, pwd=None, wait=True, check_status=False):
```

Comment:
Shouldn't "pwd" be named "stdin"? That's what it ends up being, right?

Reply:
Done.